library/src/main/java/com/proxerme/library/connection/info/entity/EntryCore.java

```diff
@@ -0,0 +1,287 @@
+package com.proxerme.library.connection.info.entity;
+
+import android.os.Parcel;
+import android.os.Parcelable;
+import android.support.annotation.IntRange;
+import android.support.annotation.NonNull;
+
+import com.afollestad.bridge.annotations.Body;
+import com.proxerme.library.parameters.CategoryParameter;
+import com.proxerme.library.parameters.GenreParameter;
+import com.proxerme.library.parameters.LicenseParameter;
+import com.proxerme.library.parameters.MediumParameter;
+
+/**
+ * Represents the core data of an Entry (Anime, Manga)
+ *
+ * @author Desnoo
+ */
+public class EntryCore implements Parcelable {
+
+    public static final Creator<EntryCore> CREATOR = new Creator<EntryCore>() {
+        @Override
+        public EntryCore createFromParcel(Parcel in) {
+            return new EntryCore(in);
+        }
+
+        @Override
+        public EntryCore[] newArray(int size) {
+            return new EntryCore[size];
+        }
+    };
+
+    @Body(name = "id")
+    String id;
+    @Body(name = "name")
+    String name;
+    @Body(name = "genre")
+    String genre;
+    @Body(name = "fsk")
+    String fsk;
+    @Body(name = "description")
+    String description;
+    @Body(name = "medium")
+    String medium;
+    @Body(name = "count")
+    int count;
+    @Body(name = "state")
+    int state;
+    @Body(name = "rate_sum")
+    int rateSum;
+    @Body(name = "rate_count")
+    int rateCount;
+    @Body(name = "clicks")
+    int clicks;
+    @Body(name = "category")
+    String category;
+    @Body(name = "license")
+    int license;
+
+    /**
+     * Parser to retain the instance back from the parcel.
+     *
+     * @param in the parcel to parse.
+     */
+    protected EntryCore(Parcel in) {
+        id = in.readString();
+        name = in.readString();
+        genre = in.readString();
+        fsk = in.readString();
+        description = in.readString();
+        medium = in.readString();
+        count = in.readInt();
+        state = in.readInt();
+        rateSum = in.readInt();
+        rateCount = in.readInt();
+        clicks = in.readInt();
+        category = in.readString();
+    }
+
+    /**
+     * Returns the id of this entry.
+     *
+     * @return The id.
+     */
+    @NonNull
+    public String getId() {
+        return id;
+    }
+
+    /**
+     * Returns the name of this entry.
+     *
+     * @return the name.
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Returns the genre of this entry.
+     *
+     * @return The genre.
+     */
+    @NonNull
+    @GenreParameter.Genre
+    public String getGenre() {
+        return genre;
+    }
+
+    /**
+     * Returns an array of fsk names of this entry.
+     *
+     * @return An array of fsk names.
+     */
+    @NonNull
```

Comment:
We could annotate this with the FskConstraint annotation, to indicate that this array only can include those values.